### PR TITLE
feat: add configure-grub just command

### PR DESCRIPTION
**Pull request by sekalengrengginang**
based on [bazzite configure grub just command](https://github.com/ublue-os/bazzite/blob/main/system_files/desktop/shared/usr/share/ublue-os/just/80-bazzite.just#L44) with command from [hidden grub changeset](https://fedoraproject.org/wiki/Changes/HiddenGrubMenu)

### CODE REVIEW

**Comment by bketelsen** [comment]
i'll let others comment here too, but I think we're trying to reduce the amount of inline bash scripting in our just files and instead put an actual script on disk and call it from just.

**Comment by sekalengrengginang** [comment]
> i'll let others comment here too, but I think we're trying to reduce the amount of inline bash scripting in our just files and instead put an actual script on disk and call it from just.

done, is /usr/libexec the correct location for the script?

File: system_files/shared/usr/libexec/configure-grub.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+source /usr/lib/ujust/ujust.sh
+GRUB_STATE="$(sudo grub2-editenv list | grep "^menu_auto_hide=")"
+if [ "$GRUB_STATE" == "menu_auto_hide=0" ] ; then
```
Comment (m2Giles) [inline system_files/shared/usr/libexec/configure-grub.sh:4]: This isn't set on older installs. If the variable doesn't exist it is also not hidden.

File: system_files/shared/usr/libexec/configure-grub.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    GRUB_STATE="${b}Not Hidden${n}"
+else
+    GRUB_STATE="${b}Hidden${n}"
+fi
+echo "${bold}Grub menu configuration${normal}"
+echo "Grub menu is set to: $GRUB_STATE"
+OPTION=$(Choose "Hide Grub" "Unhide Grub" )
+if [[ "${OPTION,,}" =~ ^hide ]]; then
+    sudo grub2-editenv - set menu_auto_hide=2 boot_success=1
```
Comment (m2Giles) [inline system_files/shared/usr/libexec/configure-grub.sh:13]: menu_auto_hide=1 should hide the boot_menu so long as there was a successful shutdown.

Don't set boot_success=1. That variable should only be set when the computer successfully powers down.

**Review by m2Giles — CHANGES_REQUESTED** [review]
Quick comments. menu_auto_hide=2 always hides. Additionally your condition check will say the grub menu is hidden even when it's not.

I think you should have 3 states.

Always Hide, Auto Hide, and Not Hidden. Auto Hide should be the default behaviour on new installs.

Thank you for adding this, and it will be a great place to put the rest of grub logic like removing duplicate entries and regenerating grub.cfg

File: system_files/shared/usr/libexec/configure-grub.sh (hunk 1 above)
Comment (sekalengrengginang) [reply system_files/shared/usr/libexec/configure-grub.sh:4]: it should exist since Fedora 29, did bluefin remove those env variable?

Comment (m2Giles) [reply system_files/shared/usr/libexec/configure-grub.sh:4]: They weren't on my original 37 and 38 silverblue installs. Unsure why.

But it's besides the point if the variable doesn't exist it's not hidden which is what your code currently says.

**Review by m2Giles — APPROVED** [review]

**Comment by sekalengrengginang** [comment]
thank you for adding check for grub state

**Pull request merged**